### PR TITLE
Adding navbar routing views

### LINKED ISSUE #37: View navigation
### Description:
As a UI guru, I want to be able to navigate to different views so that I can explore the program more.

### Assumptions:
* Main window is initialized

### Estimate:
* 5

### Tasks:
* Make a widget on the side of the window for holding buttons
* Connect those buttons to open different views

### Tests:
* Add views and navigate to them

### Definition of Done:
The user is able to switch to different views by the click of buttons on the side.

### PR DESCRIPTION
**Pull request by Roger-Chavez**
Closes #37 

### Key features
* The program now has a navbar and is able to navigate to all views 
* Made a background for dashboard
### Future Improvements
* Views plugged into the stack widgets will need to pass in their resets to the main-window reset

### Notes
none

### CODE REVIEW

File: src/widgets/navbar.cpp [hunk 1]
```diff
@@ -0,0 +1,156 @@
+#include "navbar.hpp"
+#include "navitem.hpp"
+#include <QPropertyAnimation>
+
+/**
+ * @brief Constructor
```
Comment (ChrisRzech) [inline src/widgets/navbar.cpp]: Useless brief, remove it

File: src/widgets/navbar.cpp [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    QPalette palette;
+    palette.setColor(QPalette::Disabled, QPalette::Highlight, Qt::darkBlue);
+    palette.setColor(QPalette::Inactive, QPalette::Highlight, Qt::darkBlue);
+    palette.setColor(QPalette::Active, QPalette::Highlight,  Qt::darkBlue);
+    this->setPalette(palette);
+
+}
+
+/**
+ * @brief Set height of the bar
```
Comment (ChrisRzech) [inline src/widgets/navbar.cpp]: Useless brief, remove it

File: src/widgets/navbar.cpp [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+ * @param height The new height of the bar and its parent
+ */
+void NavBar::setHeight(int height)
+{
+    resize(width(), height);
+    parentWidget()->resize(width(), height);
+}
+
+/**
+ * @brief Set minimum width of the bar
```
Comment (ChrisRzech) [inline src/widgets/navbar.cpp]: Useless brief, remove it

File: src/widgets/navbar.cpp [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+ *
+ * @param width The new minimum width
+ */
+void NavBar::setMinWidth(int width)
+{
+    m_minWidth = width;
+}
+
+/**
+ * @brief Set maximum width of the bar
```
Comment (ChrisRzech) [inline src/widgets/navbar.cpp]: Useless brief, remove it

File: src/windows/mainwindow.ui [hunk 5]
```diff
@@ -1,24 +1,138 @@
...
- <widget class="QMainWindow" name="MainWindow" >
-  <property name="geometry" >
+ <widget class="QMainWindow" name="MainWindow">
+  <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
-    <width>400</width>
-    <height>300</height>
+    <width>710</width>
```
Comment (ChrisRzech) [inline src/windows/mainwindow.ui]: Change window width to 800

**Review by ChrisRzech — CHANGES_REQUESTED** [review]
Documentation changes and window size change

File: src/widgets/navbar.cpp (hunk 1 above)
Comment (Roger-Chavez) [reply src/widgets/navbar.cpp]: ok, I'll remove the comment you wrote

File: src/widgets/navbar.cpp (hunk 2 above)
Comment (Roger-Chavez) [reply src/widgets/navbar.cpp]: ok, I'll remove the comment you wrote

File: src/widgets/navbar.cpp (hunk 3 above)
Comment (Roger-Chavez) [reply src/widgets/navbar.cpp]: ok, I'll remove the comment you wrote

File: src/widgets/navbar.cpp (hunk 4 above)
Comment (Roger-Chavez) [reply src/widgets/navbar.cpp]: ok, I'll remove the comment you wrote

File: src/widgets/navbar.hpp [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+ *
+ * @see <a href="https://fontawesome.com/icons?d=gallery&m=free">Font Awesome icon reference</a>
+ */
+class NavBar : public QListWidget
+{
+    Q_OBJECT
+
+public:
+    /* Constructor */
+    NavBar(QWidget* parent, int minWidth, int maxWidth);
```
Comment (ChrisRzech) [inline src/widgets/navbar.hpp]: Can we get the `explicit` keyword back on this? This will allow us to not put a `NavBar` anywhere else besides a `QWidget`

Comment (Roger-Chavez) [reply src/widgets/navbar.hpp]: k

**Review by ChrisRzech — APPROVED** [review]
Ez copy and paste 👍

**Review by WakkaTrout — APPROVED** [review]
NavBar code looks good. I don't see any bugs

**Pull request merged**